controllers/configurationpolicy_controller.go

```diff
@@ -2556,6 +2558,30 @@ func handleSingleKey(
 			mergedValue.(map[string]interface{}), existingValue.(map[string]interface{}))
 	}
 
+	if key == "stringData" && existingObj.GetKind() == "Secret" {
+		// override automatic conversion from stringData to data prior to evaluation
+		existingValue = existingObj.UnstructuredContent()["data"]
```

Comment:
Optional/FYI: There's an `unstructured` function, `unstructured.NestedStringMap()` that could replace the type assertions and grab the `map[string]string` directly, though it's not a huge benefit here.
```
encodedValue, _, err := unstructured.NestedStringMap(existingObj.Object, "data")
```